tools/c7n_azure/c7n_azure/resources/sqldatabase.py

```diff
@@ -238,3 +271,174 @@ def get_retention_from_backup_policy(self, retention_policy):
+                    retention-period-units: months
+
+    """
+
+    schema = type_schema(
+        'update-long-term-backup-retention-policy',
+        rinherit=LongTermBackupRetentionPolicyFilter.schema,
+        op=None
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(LongTermBackupRetentionPolicyAction, self).__init__(
+            BackupRetentionPolicyHelper.LONG_TERM_SQL_OPERATIONS, *args, **kwargs)
+        self.backup_type = BackupRetentionPolicyHelper.LongTermBackupType[self.data.get(
+            'backup-type')]
+        retention_period = self.data['retention-period']
+        retention_period_units = RetentionPeriod.Units[self.data['retention-period-units']]
+        self.iso8601_duration = RetentionPeriod.iso8601_duration_from_period_and_units(
+            retention_period,
+            retention_period_units
+        )
+
+    def get_parameters_for_new_retention_policy(self, database):
+        client = self.manager.get_client()
+        get_operation = getattr(client, self.operations_property).get
+        current_retention_policy = BackupRetentionPolicyHelper.get_backup_retention_policy(database,
+            get_operation, self.operations_property)
+
+        new_retention_policy = self._copy_retention_policy(current_retention_policy)
+        new_retention_policy[self.backup_type.retention_property] = self.iso8601_duration
+
+        # Make sure that the week_of_year is set properly based on what
+        # the yearly backup retention is. If this is not done, the API will
+        # fail with an invalid parameter value
+        yearly_retention = new_retention_policy[
+            BackupRetentionPolicyHelper.LongTermBackupType.yearly.retention_property
+        ]
+        week_of_year = new_retention_policy[BackupRetentionPolicyHelper.WEEK_OF_YEAR]
+        if yearly_retention is None:
+            # Without a yearly retention, the week should be 0
+            new_retention_policy[BackupRetentionPolicyHelper.WEEK_OF_YEAR] = 0
+        elif not week_of_year:
+            # If there is a yearly retention, and the week is not specified, default to week 1
+            new_retention_policy[BackupRetentionPolicyHelper.WEEK_OF_YEAR] = 1
+
+        return BackupLongTermRetentionPolicy(**new_retention_policy)
+
+    def _copy_retention_policy(self, retention_policy):
+        """
+        Create a copy of a retention policy object with only the required parameters for the
+        BackupLongTermRetentionPolicy class.
+
+        more info:
+          https://docs.microsoft.com/en-us/python/api/azure-mgmt-sql/azure.mgmt.sql.models.backuplongtermretentionpolicy?view=azure-python
+        """
+
+        new_retention_policy = {}
+        for backup_type in BackupRetentionPolicyHelper.LongTermBackupType:
+            key = backup_type.retention_property
+
+            # Even though the api returns `PT0S` as a value for "zero duration", it does not accept
+            # that as a valid input parameter value.
```

Comment:
Can you open an issue for the API? Seems like a wrong behavior

Reply:
yeah, I agree

Reply:
After playing a bit more with this, I think it was actually operator error on my part. I was testing with the yearly retention, where the week of year needed to also be specified. This case is handled already in `get_parameters_for_new_retention_policy` where I check to see if the yearly retention is specified. 

Removing this special case from the code works.